main_app/views.py

```diff
@@ -0,0 +1,37 @@
+from django.shortcuts import render
+from django.views.generic import TemplateView
+
+from .igdb_api import IGDB
+
+
+class MainPageView(TemplateView):
+    template = 'main_page.html'
+
+    def __init__(self, **kwargs):
+        super(MainPageView, self).__init__(**kwargs)
+        self.client = IGDB(6)
+        self.__current_page = '1'
+        self.__search = ''
+        self.__platforms = ''
+        self.__genres = ''
+        self.__user_ratings = ''
+
+    def get_context_data(self, **kwargs):
+        super(MainPageView, self).get_context_data()
+        self.__current_page = self.request.GET.get('page', self.__current_page)
+        print(self.request.GET)
+        games = self.client.api_get_games_list(**{key: value for key, value in self.request.GET.items()})
+        pages_amount = self.client.api_get_last_pages_amount()
```

Comment:
Declare your `client` variable just inside `get_context_data` instead of __ init __.